nemo/collections/asr/parts/submodules/subsampling.py

```diff
@@ -145,30 +163,50 @@ def __init__(self, subsampling, subsampling_factor, feat_in, feat_out, conv_chan
                 in_channels = conv_channels
 
         elif subsampling == 'striding':
-            self._padding = 1
             self._stride = 2
             self._kernel_size = 3
             self._ceil_mode = False
 
+            if self.is_causal:
+                self._left_padding = self._kernel_size - 1
+                self._right_padding = self._stride - 1
+                self._max_cache_len = subsampling_factor + 1  # TODO: does it work for all cases?
```

Comment:
Is this tested?

Reply:
Yes, it is fine. I checked it already. Dropped the TODO.